### PR TITLE
Use dataclasses for NodeIdentifier

### PR DESCRIPTION
**Pull request by henriquegemignani**
Negligible time difference, but actually a bit faster. Means we can also not depend on `attrs` (but it's still installed as transient dependency)

```
> randovania development benchmark compare tools\benchmark-7.5.0.json .\tools\benchmark-nodeidentifier.json
                    Difference | Fixes | Failures |     Mean |    Stdev |   Median
        Blank Development Game |     0 |        0 |   -0.004 |    0.007 |   -0.003
                 Metroid Prime |     0 |        0 |    0.016 |    0.066 |    0.026
       Metroid Prime 2: Echoes |     0 |        0 |   -0.035 |    0.126 |   -0.037
   Metroid Prime 3: Corruption |     0 |        0 |   -0.058 |    0.115 |   -0.040
                 Super Metroid |     0 |        0 |   -0.031 |    0.084 |   -0.026
                 Metroid Dread |     0 |        0 |   -0.132 |    0.349 |   -0.097
        Metroid: Samus Returns |     0 |        0 |    0.049 |    0.562 |    0.015
                    Cave Story |     0 |        0 |   -0.002 |    0.099 |   -0.018
      Another Metroid 2 Remake |     0 |        0 |    0.048 |    0.158 |    0.080
                Metroid Fusion |     0 |        0 |    0.030 |    0.131 |    0.022
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -53,9 +53,6 @@ dependencies = [
     # nonblocking file IO
     "aiofiles",
 
-    # Better than dataclasses for NodeIdentifier
```
Comment (duncathan) [inline pyproject.toml:56]: lmao

Comment (henriquegemignani) [reply pyproject.toml:56]: It _probably_ was faster in 3.10 or 3.9 when I did this

Comment (duncathan) [reply pyproject.toml:56]: oh I'm sure, just very funny

File: test/games/am2r/test_am2r.py [hunk 2]
```diff
@@ -66,8 +66,8 @@ def test_all_tricks_should_have_proper_requirements():
                 if len(set(required_resource_collection.values())) > 1:
                     database_dict[required_resources].append(
                         (
-                            f"{node.name} ({node.identifier.area_name})",
-                            f"{dest_node.name} ({dest_node.identifier.area_name})",
+                            f"{node.name} ({node.identifier.area})",
```
Comment (Miepee) [inline test/games/am2r/test_am2r.py:69]: mypy

Comment (henriquegemignani) [reply test/games/am2r/test_am2r.py:69]: I'm not touching this one :p

**Pull request merged**